.travis.yml

```diff
@@ -44,7 +44,7 @@ script:
   - if [ "$DDS_SECURITY" == 1 ]; then BUILD_TARGETS+="OpenDDS_Security"; fi
   - make -C OpenDDS -sj6 DCPSInfoRepo_Main OpenDDS_Rtps_Udp $BUILD_TARGETS
   - npm install
-  - export V8_ROOT=~/.node-gyp/$(node-gyp list)
+  - export V8_ROOT=$(node-gyp list -v |& grep dir\: | cut -d ':' -f 2 | sed 's/^[ ]* //g')
```

Comment:
Looking at the [node-gyp docs](https://github.com/nodejs/node-gyp#configuration), it looks like we could just manually set this path.

So maybe something like:

```yml
env:
  global:
    - npm_config_devdir=$TRAVIS_BUILD_DIR/opendds-node-gyp-devdir
    - V8_ROOT=$npm_config_devdir
```

Reply:
Weird. If you specify the path, it appends the node version. If you don't, it doesn't? Or maybe it's just checking to see if the version is somewhere in the path?

Reply:
I'm not sure. Yesterday I just did a check that `npm_config_devdir` would create the directory if I ran `node-gyp install --ensure`. I didn't actually see what was in it. Looking at it now, my manual one is nested in a version directory too. My normal `.node-gyp` also has versions in it too, but it might be because I have multiple installations of node on my system.

I think what you changed it to is acceptable though. `node -v` should be less brittle than parsing everything that `node-gyp list -v` outputs, which is what I was worried about.

Reply:
I think older versions used ~/.node-gyp, but newer ones used ~/.cache/.node-gyp or something similar. MacOS put it in ~/Library/Caches/node-gyp. Too many "magic" locations. I was having trouble because node-gyp was returning two versions (from previous state) but the newer one (v10.22.0) was second in the list and so the wrong version was being passed into the path. I eventually cleaned it all out and was building locally with a fresh directory / environment, but it took a while to root out all the sneaky cached locations. I agree that node -v should hopefully be more predictable.

Reply:
Also, we're periodically getting test failures because the inforepo doesn't like being started / stopped so quickly.